cmd/controller/controller.go

```diff
@@ -189,12 +191,50 @@ func (c *Controller) unseal(key string) error {
 	}
 
 	_, err = c.sclient.Secrets(ssecret.GetObjectMeta().GetNamespace()).Create(secret)
-	if err != nil && errors.IsAlreadyExists(err) {
-		_, err = c.sclient.Secrets(ssecret.GetObjectMeta().GetNamespace()).Update(secret)
+	if err != nil && !errors.IsAlreadyExists(err) {
+		// Error wasn't already exists so is real error
+		return err
 	}
+
```

Comment:
On a fresh create, I think you're now doing a Create followed by an immediate Update.

I think we need a `if err == nil { return nil }` about here (or equivalent change to the above predicate) to avoid the unnecessary update.

Reply:
That should do it